src/seedu/addressbook/data/person/Address.java

```diff
@@ -21,36 +138,51 @@
      * @throws IllegalValueException if given address string is invalid.
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
-        String trimmedAddress = address.trim();
-        this.isPrivate = isPrivate;
-        if (!isValidAddress(trimmedAddress)) {
+        if (!isValidAddress(address.trim())) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+        setAddressParams(isPrivate, address.trim());
     }
 
     /**
-     * Returns true if a given string is a valid person email.
+     * Returns true if a given string is a valid person address.
      */
     public static boolean isValidAddress(String test) {
-        return test.matches(ADDRESS_VALIDATION_REGEX);
+        return ADDRESS_VALIDATION_REGEX.matcher(test).matches();
+    }
+
+    /**
+     * Splits address into block, street, unit and postal code
+     * 
+     * @throws IllegalValueException if any of the above have invalid format
+     */
+    private void setAddressParams(boolean isPrivate, String addressParams) throws IllegalValueException {
+        this.isPrivate = isPrivate;
+        final Matcher matcher = ADDRESS_VALIDATION_REGEX.matcher(addressParams);
+        matcher.matches();
```

Comment:
Be careful when using Regex.
In this case what happens when `matcher.matches()` returns a false?
What happens if the address provided does not fit your Regex?